state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -120,6 +126,19 @@ pub enum DepositIgnoredReason {
 	/// The deposit was ignored because the amount provided was not high enough to pay for the fees
 	/// required to process the requisite transactions.
 	NotEnoughToPayFees,
+	TransactionTainted,
+}
+
+/// Holds information about a tainted transaction.
+#[derive(RuntimeDebug, PartialEq, Eq, Encode, Decode, TypeInfo, CloneNoBound)]
+#[scale_info(skip_type_params(T, I))]
+pub struct TaintedTransactionDetails<T: Config<I>, I: 'static> {
+	/// The broker that created the tainted transaction.
+	pub broker: T::AccountId,
```

Comment:
nit: it's not always going to be a broker.

```suggestion
	/// The account that reported the transaction.
	pub reporter: T::AccountId,
```

Reply:
(But also, i think we can remove this field altogether? See: https://github.com/chainflip-io/chainflip-backend/pull/5310#discussion_r1795051064